static/app/views/performance/queues/messageSamplesPanel.tsx

```diff
@@ -168,36 +188,13 @@ export function MessageConsumerSamplesPanel() {
           </ModuleLayout.Full>
 
           <ModuleLayout.Full>
-            <MetricsRibbon>
-              <MetricReadout
-                align="left"
-                title={t('Processed')}
-                value={transactionMetrics?.[0]?.['count()']}
-                unit={'count'}
+            <MetricsRibbonContainer>
+              <MetricsRibbon
+                type={messageActorType}
+                metrics={transactionMetrics}
                 isLoading={aretransactionMetricsFetching}
               />
```

Comment:
IMO would be just as good, if not better, to do the conditional ribbon here, rather than making a new `MetricsRibbon` component!

```
{messageActorType === MessageActorType.PRODUCER ? <ProducerMetricsRibb...
```

Otherwise, it's ribbons 3 levels down, which is a lot of levels

Reply:
Simplified!